### DISCUSSION TITLE
Webhook Support for real-time updates

### DISCUSSION THREAD

**Discussion by iklein99** [iklein99/homebridge-smartthings#66] [Announcements]
Hi all,

I am thinking about implementing a service which will allow you to get near real-time updates from devices like sensors.  So, for instance, a motions sensor today using this plugin may take around 5 seconds or so to trigger an automation due to the way the plugin gets the update via a polling algorithm. 

This new feature will allow instant updates for all sorts of devices like locks and doors and other sensors.  This will require me to maintain cloud compute resources to receive these updates and send them onto your home bridges.  This will cost me $$ and I will have to pass that cost on to you.  

I'm asking if this is something the community would want and support.  It won't pay if I only have a few takers.

Please reply to this and let me know your thoughts.

**Comment by coryabooth** [comment]
I would be interested, couple thoughts though:

1. It may be a bit early to determine how popular this may become as I myself just recently began looking at potential alternatives to Groovy.  None of my 95 devices have been force migrated yet.  I am sure there will be a good portion of people looking around soon, once things stop working as configured.
2. Hopefully this will be a solution that can be easily retracted?  Like almost everything in the smart-home space, I get a little concerned with putting too much faith in a particular solution to stay "as-is" for an extended period of time.  But if this was to work as described, I could see people building extensive automation routines that would take days to replace if this was to stop working.
3.  Edge...  The big selling point of Edge is to have "near real time" reaction from devices.  If this holds true, I would assume everyone would still leverage SmartThings for the majority of their automation needs?

However, I would gladly pay a fee for this.  I have put a lot of money into smart home devices and, even if this was to last a couple of years (hopefully longer), it would be worth it to me.
